src/classes/User.as

```diff
@@ -229,6 +236,77 @@ package classes
             this.averageRank = (rankTotal / _gvars.TOTAL_PUBLIC_SONGS);
         }
 
+        ///- Determine what songs make up the user's Skill Rating
+        public function getUserSkillRatingData():void
+        {
+            for (var key:int in this.level_ranks)
+            {
+                var levelRank:Object = this.level_ranks[key];
+                
+                //Calculate the AAA Equiv for the scores on the song if greater than 0
+                if (levelRank.score != 0)
+                {
+                    //Get the song's difficulty
+                    var songInfo:SongInfo = _playlist.getSongInfo(key);
+                    
+                    //Calculate the song's equiv
+                    levelRank.equiv = SkillRating.calcSongWeightFromScore(levelRank.rawscore, songInfo);
+
+                    //Add it to the Skill Rating list if it's not 0
+                    if (levelRank.equiv > 0)
+                        skill_rating_songs.push(levelRank);
+                }
+            }
+
+            //Sort based on equiv
+            skill_rating_songs.sort(equivSort);
+
+            //Dump all but the top X
+            while (skill_rating_songs.length > skill_rating_top_count)
```

Comment:
This can be simplified to just:
```
if(skill_rating_songs.length > skill_rating_top_count)
    skill_rating_songs.length = skill_rating_top_count;
```

This will trim excess items of the array instead of 1 at a time.

Reply:
Fixed, cheers for that.